src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -853,43 +853,60 @@ def generate_flow_tools_json(
     return flow_tools
 
 
-def update_user_agent_from_env_var():
-    """Update user agent from env var to OperationContext"""
-    from promptflow._core.operation_context import OperationContext
+class ClientUserAgentUtil:
+    """SDK/CLI side user agent utilities."""
+
+    @classmethod
+    def _get_context(cls):
+        from promptflow._core.operation_context import OperationContext
+
+        return OperationContext.get_instance()
+
+    @classmethod
+    def get_user_agent(cls):
+        context = cls._get_context()
+        # directly get from context since client side won't need promptflow/xxx.
+        return context.get(context.USER_AGENT_KEY, "").strip()
+
+    @classmethod
+    def append_user_agent(cls, user_agent: Optional[str]):
+        if not user_agent:
+            return
+        context = cls._get_context()
+        context.append_user_agent(user_agent)
+
+    @classmethod
+    def update_user_agent_from_env_var(cls):
+        for env_name in [USER_AGENT, PF_USER_AGENT]:
+            if env_name in os.environ:
+                cls.append_user_agent(os.environ[env_name])
+
 
-    if USER_AGENT in os.environ:
-        # Append vscode or other user agent from env
-        OperationContext.get_instance().append_user_agent(os.environ[USER_AGENT])
+def get_client_user_agent():
+    """Get client user agent from OperationContext.
 
-    if PF_USER_AGENT in os.environ:
-        # Append promptflow user agent from env
-        OperationContext.get_instance().append_user_agent(os.environ[PF_USER_AGENT])
+    For client scenario, ua won't include promptflow/xxx.
+
+    """
+    return ClientUserAgentUtil.get_user_agent()
 
 
-def setup_user_agent_to_operation_context(user_agent=None):
+def setup_user_agent_to_operation_context(user_agent):
     """Setup user agent to OperationContext.
     For calls from extension, ua will be like: prompt-flow-extension/ promptflow-cli/ promptflow-sdk/
     For calls from CLI, ua will be like: promptflow-cli/ promptflow-sdk/
     For calls from SDK, ua will be like: promptflow-sdk/
     """
-    from promptflow._core.operation_context import OperationContext
-
-    update_user_agent_from_env_var()
-    # Append user agent
-    context = OperationContext.get_instance()
-    # skip append if empty
-    if user_agent:
-        context.append_user_agent(user_agent)
-    return context.get_user_agent()
+    ClientUserAgentUtil.update_user_agent_from_env_var()
+    ClientUserAgentUtil.append_user_agent(user_agent)
```

Comment:
Add the SDK/CLI agent first, and then add the agent in the user env?

Reply:
Will the order impact behavior?